cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -55,19 +54,24 @@ func (s *server) Start(ctx context.Context, host component.Host) error {
 	mf := otelmetrics.NewFactory(s.telset.MeterProvider)
 	baseFactory := mf.Namespace(metrics.NSOptions{Name: "jaeger"})
 	queryMetricsFactory := baseFactory.Namespace(metrics.NSOptions{Name: "query"})
-	f, err := jaegerstorage.GetStorageFactory(s.config.Storage.TracesPrimary, host)
+	v1Factory, err := jaegerstorage.GetStorageFactory(s.config.Storage.TracesPrimary, host)
 	if err != nil {
-		return fmt.Errorf("cannot find primary storage %s: %w", s.config.Storage.TracesPrimary, err)
+		return fmt.Errorf("cannot find primary storage from v1 factory %s: %w", s.config.Storage.TracesPrimary, err)
+	}
+	v2Factory, err := jaegerstorage.GetStorageFactoryV2(s.config.Storage.TracesPrimary, host)
+	if err != nil {
+		return fmt.Errorf("cannot find primary storage from v2 factory %s: %w", s.config.Storage.TracesPrimary, err)
 	}
 
-	spanReader, err := f.CreateSpanReader()
+	v1Factory.CreateSpanReader()
+	traceReader, err := v2Factory.CreateTraceReader()
 	if err != nil {
 		return fmt.Errorf("cannot create span reader: %w", err)
 	}
 
-	spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, queryMetricsFactory)
+	// TODO: decorate trace reader with metrics
```

Comment:
@yurishkuro How should we do this? The [decorator](https://github.com/jaegertracing/jaeger/blob/main/storage/spanstore/metrics/decorator.go) implements `spanstore.Reader` so if we create a new one on `spanstore_v2.Reader`, it won't work currently since we only ever call the underlying v1 methods. The other alternative could be is that we take we pass `metrics.Factory` (or `otel.MeterProvider`) into the factoryadapter and decorate the `spanReader` before wrapping it into a TraceReader. Let me know what you think.

Reply:
I would like to defer this if possible - what if we use the existing v1 decorator before creating V2 adapter? I.e. I would wrap the span reader as soon as we get it from the v1 factory.

Reply:
@yurishkuro Do we still want to use `GetStorageFactoryV2`? If we do, then the tracereader is created from the spanreader there. The other approach would be to just initialize the trace reader here (which is what I was previously doing) using the factoryadapter rather than getting it from the v2 factory. What do you think?